### PR TITLE
address ambiguous auto variable.

### PR DESCRIPTION
**Pull request by fujitatomoya**
address https://github.com/ros2/rclcpp/issues/1963 and https://github.com/dusty-nv/jetson-containers/issues/336

### CODE REVIEW

**Review by fujitatomoya — COMMENTED** [review]
originally this was tried to be fixed with https://github.com/ros2/rclcpp/pull/1976, but it got lost.

CC: @clalancette @alsora

File: rclcpp/include/rclcpp/client.hpp [hunk 1]
```diff
@@ -864,7 +864,7 @@ class Client : public ClientBase
         "Received invalid sequence number. Ignoring...");
       return std::nullopt;
     }
-    auto value = std::move(it->second.second);
+    auto value = std::optional<CallbackInfoVariant>{std::move(it->second.second)};
```
Comment (alsora) [inline rclcpp/include/rclcpp/client.hpp:867]: why not
```suggestion
    std::optional<CallbackInfoVariant> value = std::move(it->second.second);
```

Comment (fujitatomoya) [reply rclcpp/include/rclcpp/client.hpp]: that is better, thanks!

**Review by alsora — APPROVED** [review]

**Comment by fujitatomoya** [comment]
Windows is unstable?

* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21429/)

**Comment by fujitatomoya** [comment]
@Mergifyio backport humble iron

**Pull request merged**